clang/lib/Driver/ToolChains/Arch/Sparc.cpp

```diff
@@ -179,6 +179,13 @@ void sparc::getSparcTargetFeatures(const Driver &D, const ArgList &Args,
       Features.push_back("-hard-quad-float");
   }
 
+  if (Arg *A = Args.getLastArg(options::OPT_mv8plus, options::OPT_mno_v8plus)) {
+    if (A->getOption().matches(options::OPT_mv8plus))
```

Comment:
We usually know whether a feature is default on or off.
If it is default off, the driver only passes `+xxx`, but suppresses `-xxx`.

Are both `+v9` and `-v9` needed here?

Reply:
This is quite confusing.
`-m[no]-v8plus` is an environment option, similar to X86's `-mx32`, but imposes more restrictions.
`-mcpu=v9 -mno-v8plus` should be equivalent to just `-mcpu=v9` because 64-bit environment is the default for 64-bit ISA.
`-mv8plus` enables 32-bit environment and therefore should restrict the compiler from using the whole set of V9 features. That is, I would expect it to disable something rather than to enable `v9`, which should already be set by `-mcpu=v9`.

Reply:
Either way, the order of `-m8plus` relative to `-mcpu` shouldn't matter (one sets the environment, the other sets an ISA).

Reply:
> on GCC, `-m[no-]v8plus` also implies that V9 instructions are available (or not, in case of `-mno-v8plus`)

Well, no. Not exactly. Whether or not V9 instructions are available is determined strictly by `-mcpu=v9` (which is the default). In both cases by the first godbolt link instructions are V9 instructions and operate on 64-bit registers.
The thing is, you can't use the upper half of these registers, which makes instructions such as MULX useless. There are, however, other instructions introduced in V9 that are useful -- 32-bit atomics for instance.

Also note that `-mno-v8plus` is the default and passing it explicitly won't change the behavior (unless you specified `-mv8plus` prior to it).

Reply:
> In that case, would making these a no-op flag okay for now? As far as I understand it that would still be a compliant implementation, no?

AFAIK it is an ABI affecting flag, although I don't know if it changes anything except for ELF header's e_machine field compared to 32-bit V8/V9. It can't safely be ignored if we claim support for V8+. If we don't, and V8+ is otherwise compatible with 32-bit ABI, I think ignoring it and generating  instructions should be fine, probably with a warning.

> Codegen, etc. changes will happen in future patches but I can amend this one to include a placeholder `v8plus` feature bit in the backend.
> 
> What do you think about it?

I have little experience in target feature bits. It could be helpful to have a draft that adds some V8+ support to the backend and the corresponding feature bits.